soundmeter/meter.py

```diff
@@ -124,27 +132,64 @@ def start(self):
+                    soundvalue = self.weightedvalue(segment, self.weightingcurve)
+                else:
+                    sys.exit(1)  # Value type must be either db or rms
                 if self.collect:
-                    self.collect_rms(rms)
-                self.meter(rms)
+                    self.collect_rms(soundvalue)
+                self.meter(soundvalue)
                 if self.action:
-                    if self.is_triggered(rms):
-                        self.execute(rms)
-                self.monitor(rms)
+                    if self.is_triggered(soundvalue):
+                        self.execute(soundvalue)
+                self.monitor(soundvalue)
             self.is_running = False
             self.stop()
 
         except self.__class__.StopException:
             self.is_running = False
             self.stop()
 
-    def meter(self, rms):
+    def meter(self, value):
         if not self._graceful:
-            sys.stdout.write('\r%10d  ' % rms)
+            sys.stdout.write('\r%10d  %s' % (value, self.valuestype))
             sys.stdout.flush()
             if self.log:
-                self.logging.info(rms)
+                self.logging.info(value)
+
+    def weightedvalue(self, segment, curve):
+        """
+        This function it is used to calculate the average dB level using the weighting curves.
+        The zeros and poles were created by using those functions:
+        http://siggigue.github.io/pyfilterbank/splweighting.html
+        with the sample rate hardcoded to 44100Hz
+        More info about weighting curves:
+        https://www.cirrusresearch.co.uk/blog/2011/08/what-are-a-c-z-frequency-weightings/
+        :param segment: The segment to calculate the weighted average
+        :param curve: The type of curve to apply Z, A, B or C
+        :return: The weighted average
+        """
+        if curve.lower() == "z":
+            return segment.dBFS
+        if curve.lower() == "a":
+            b = np.array([0.25574113, -0.51148225, -0.25574113, 1.0229645, -0.25574113,
+                    -0.51148225, 0.25574113])
+            a = np.array([1.00000000e+00, -4.01957618e+00, 6.18940644e+00,
+                          -4.45319890e+00, 1.42084295e+00, -1.41825474e-01,
+                          4.35117723e-03])
+        if curve.lower() == "b":
+            b = np.array([0.21727294, -0.21727294, -0.43454587, 0.43454587, 0.21727294,
+                          -0.21727294])
+            a = np.array([1., -3.11234468, 3.36634059, -1.40032549, 0.15112883,
+                          -0.00479909])
+        if curve.lower() == "c":
+            b = np.array([0.21700856, 0., -0.43401712, 0., 0.21700856]),
+            a = np.array([1., -2.13467496, 1.27933353, -0.14955985, 0.0049087])
+        y = np.float32(lfilter(b, a, segment.get_array_of_samples()))
+        audio_segment = pydub.AudioSegment(y.tobytes(), frame_rate=44100, sample_width=y.dtype.itemsize, channels=1)
```

Comment:
Shall we pull the config for frame_rate from `self.config.RATE`?

Reply:
You're right! But for that, we should have weighting curves that depends on frequency.